### PR TITLE
add data processing page

### PR DESCRIPTION
**Pull request by mshodge**
Have changed the following pages to:

- [Data FAQs](https://user-guidance.analytical-platform.service.justice.gov.uk/data/data-faqs/index.html) - to try and better define different types of data

Then added the page:

- Tools for Data Processing Pipelines - sits under just Tools (as didn't fit in elsewhere) but referenced in [Tools index page](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/index.html) (with some other refactoring)

### CODE REVIEW

**Review by SoumayaMauthoorMOJ — COMMENTED** [review]
@Michael Hodge I’m still making a few notes but having “Tools for Data Processing Pipelines” under Core tools and services with a replication of the tools is a bit confusing. Is there a way to split up the tools for data processing away from the core tools and services? Something like:
```

Core tools and services
- Control Panel
- GitHub
- Data Discovery

Data Ingest and Sharing
- Data uploader
- Data Extractor
- Ingestion

IDEs
- RStudio
- JupyterLab
- Athena Console
- VS Code

Orchestration
- Airflow
- create a derived

For guidance on. which tools to use when doing data transformation please refer to [link]

**Comment by mshodge** [comment]
Reverted CaDeT material and will setup a new PR for them @jhpyke

**Comment by mshodge** [comment]
> @michael Hodge I’m still making a few notes but having “Tools for Data Processing Pipelines” under Core tools and services with a replication of the tools is a bit confusing. Is there a way to split up the tools for data processing away from the core tools and services? Something like:
> 
> ```
> 
> Core tools and services
> - Control Panel
> - GitHub
> - Data Discovery
> 
> Data Ingest and Sharing
> - Data uploader
> - Data Extractor
> - Ingestion
> 
> IDEs
> - RStudio
> - JupyterLab
> - Athena Console
> - VS Code
> 
> Orchestration
> - Airflow
> - create a derived
> 
> For guidance on. which tools to use when doing data transformation please refer to [link]
> ```

Hey @SoumayaMauthoorMOJ changed this now as (A-Z under each and added Register my data):

```
Core Tools and Services
- Control Panel
- GitHub
- Data Discovery

Data Ingest, Curation and Sharing
- Data Extractor
- Data Uploader
- Ingestion
- Register my data

Integrated Development Environment (IDE)
- Amazon Athena Console (points to https://user-guidance.analytical-platform.service.justice.gov.uk/data/curated-databases/amazon-athena/#amazon-athena)
- JupyterLab
- RStudio
- Visual Studio Code

Orchestration
- Airflow
- Create a Derived Table

```

Thoughts?

File: source/documentation/data-docs/data-faqs/index.md [hunk 1]
```diff
@@ -1,28 +1,38 @@
 # Data FAQs
 
+## What types of data are on the Platform?
+
+Data on the Analytical Platform can largely be split into four categories:
+- `Raw`: data that has been uploaded to the Analytical Platform without any changes made to it
+- `Curated`: data that has been validated, organised, cleaned and transformed by the Data Engineers ready to be used by Analytical Platform users
```
[the remark this answers is not in the document]
Comment (mshodge) [reply source/documentation/data-docs/data-faqs/index.md:7]: My data engineering experience and knowledge is limited, could you write out what you would like here and I will paste in?

File: source/documentation/data-docs/data-faqs/index.md [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
+- `Derived`: data that has been transformed, aggegregated, reduce and turned into a data model to fit specific needs of Analytical Platform users
```
[the remark this answers is not in the document]
Comment (mshodge) [reply source/documentation/data-docs/data-faqs/index.md]: My data engineering experience and knowledge is limited, could you write out what you would like here and I will paste in?

File: source/documentation/data-docs/data-faqs/index.md [hunk 3]
```diff
@@ -1,28 +1,38 @@
...
 
 ## Where should I store my own data?
 
-Data should be stored in an s3 bucket. You can create a new s3 bucket in the control panel. Data can be uploaded manually via the AWS console (which can be accessed through the control panel) or you can write it from RStudio or JupyterLab.
+Data should be stored in an S3 bucket. You can create a new S3 bucket in the [AP Control Panel](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/control-panel.html). Data can be uploaded manually via the AWS console (which can be accessed through the control panel) or you can write it from RStudio, Visual Studio Code or JupyterLab.
```
[the remark this answers is not in the document]
Comment (mshodge) [reply source/documentation/data-docs/data-faqs/index.md:29]: Added below the header:

> ⚠️ We can only hold 1,000 S3 buckets on the Analytical Platform. Before you create a new bucket, please check whether you can re-use an existing one, or create a domain-level S3 bucket and use folders and [path specific access](https://user-guidance.analytical-platform.service.justice.gov.uk/data/amazon-s3/#path-specific-access) for project level access.

File: source/documentation/data-docs/data-faqs/index.md [hunk 4]
```diff
@@ -1,28 +1,38 @@
...
 
 If your data contains anything that could be considered personal information, you must follow guidance from the data protection team which can be found on the [intranet](https://intranet.justice.gov.uk/guidance/knowledge-information/protecting-information/privacy-reform/).
 
 ## How do I read/write data from an s3 bucket?
 
-**Python/JupyterLab**: You can read/write directly from s3 using pandas. However, to get the best representation of the column types in the resulting Pandas dataframe(s), you may wish to use [mojap-arrow-pd-parser](https://github.com/moj-analytical-services/mojap-arrow-pd-parser).
+**Python**: You can read/write directly from S3 using pandas. However, to get the best representation of the column types in the resulting Pandas dataframe(s), you may wish to use [mojap-arrow-pd-parser](https://github.com/moj-analytical-services/mojap-arrow-pd-parser).
```
[the remark this answers is not in the document]
Comment (mshodge) [reply source/documentation/data-docs/data-faqs/index.md]: Not sure I quite follow "at some point we will stop allowing people to read files directly". Does this mean people could not read directly from S3? What about unstructured data like images/text files (anything that isn't in an Athena table) surely users still need to be able to access this directly?

**Comment by mshodge** [comment]
Also @SoumayaMauthoorMOJ I kept the term 'Data processing' for now instead of 'Data transformation' based on the comments on the data science slack channel. I've tried to clarify it being a general all-encompassing term in the documents now though. I've also removed reference to _pipeline_ unless referring to it in the Airflow v CaDeT section.

**Comment by SoumayaMauthoorMOJ** [comment]
@mshodge I can still see some changes to the create-a-derived-table docs, is this PR ready for me to review or are you still working on it?

File: source/documentation/data-docs/data-faqs/index.md [hunk 5]
```diff
@@ -1,28 +1,40 @@
 # Data FAQs
 
+## What types of data are on the Platform?
+
+Data on the Analytical Platform can largely be split into four categories:
+- `Raw`: data that has been uploaded to the Analytical Platform without any changes made to it
+- `Curated`: data that has been validated, organised, cleaned and transformed by the Data Engineers ready to be used by Analytical Platform users
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/data-docs/data-faqs/index.md:7]:
```suggestion
- `Curated`: data that has been validated, deduplicated and versioned by the Data Engineers ready to be used by Analytical Platform users
```

File: source/documentation/data-docs/data-faqs/index.md [hunk 6]
```diff
@@ -1,28 +1,40 @@
...
+- `Derived`: data that has been transformed, aggregated, reduce and turned into a data model to fit specific needs of Analytical Platform users
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/data-docs/data-faqs/index.md:8]:
```suggestion
- `Derived`: data that has been [denormalized](https://en.wikipedia.org/wiki/Denormalization#:~:text=Denormalization%20is%20a%20strategy%20used,data%20or%20by%20grouping%20data.), aggregated, and turned into a data model to fit specific needs of Analytical Platform users
```

File: source/documentation/data-docs/data-faqs/index.md [hunk 7]
```diff
@@ -32,18 +44,22 @@ Databases on the AP use Amazon Athena which allow you to query data using SQL. Y
...
 
-**Python/JupyterLab**: To run queries and/or read data into a pandas DataFrame, use `pydbtools`. More details are [here](https://github.com/moj-analytical-services/pydbtools). Remember to install the latest version!
+**Python**: To run queries and/or read data into a pandas DataFrame, use `pydbtools`. More details are [here](https://github.com/moj-analytical-services/pydbtools). Remember to install the latest version!
 
-**R/RStudio**: There is currently no single recommended package for querying databases in R. There is [dbtools](https://github.com/moj-analytical-services/dbtools) which should work on the "old" platform. [Rdbtools](https://github.com/moj-analytical-services/Rdbtools) should work on the "new" platform, but is not officially supported so should be used with caution.
+**R**: There is currently no single recommended package for querying databases in R. There is [dbtools](https://github.com/moj-analytical-services/dbtools) which should work on the "old" platform. [Rdbtools](https://github.com/moj-analytical-services/Rdbtools) should work on the "new" platform, but is not officially supported so should be used with caution.
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/data-docs/data-faqs/index.md:53]:
```suggestion
**R**: There is currently no single recommended package for querying databases in R. There is [dbtools](https://github.com/moj-analytical-services/dbtools) which should work on the "old" platform. [Rdbtools](https://github.com/moj-analytical-services/Rdbtools) should work on the "new" platform. This package is maintained by the analytical platform user community.
```

Comment (SoumayaMauthoorMOJ) [reply source/documentation/data-docs/data-faqs/index.md:53]: Not sure what you mean by old and new platform?

File: source/documentation/tools/index.md [hunk 8]
```diff
@@ -4,51 +4,56 @@ The Analytical (AP) provides a range of tools, services and packages. This page
...
+Allows you to browse the databases that are available on the Analytical Platform.
 
-### [JupyterLab](jupyterlab)
-Development environment for writing Python code. For more information, see the [JupyterLab documentation](https://jupyterlab.readthedocs.io/en/latest/).
+### [GitHub](https://github.com/)
+Online hosting platform for git. Git is a distributed version control system that allows you to track changes in files, while GitHub hosts the Analytical Platform's code.
 
-### [Visual Studio Code](visual-studio-code)
-General purpose code editor. For more information, see the [Visual Studio Code documentation](https://code.visualstudio.com/docs).
+## Data Ingest, Curation and Sharing
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/tools/index.md:18]:
```suggestion
## Data Ingest
```

File: source/documentation/tools/index.md [hunk 9]
```diff
@@ -4,51 +4,56 @@ The Analytical (AP) provides a range of tools, services and packages. This page
...
-### [Data Extractor](https://github.com/ministryofjustice/data-engineering-data-extractor)
+### [Ingestion](ingestion)
+An SFTP based service that allows users to ingest data into their Analytical Platform data warehouse.
 
-Extracts data from applications, services or microservices to the Analytical Platform in a standardised way.
+### [Register my data](https://github.com/ministryofjustice/register-my-data)
+Moves data from microservices into the Analytical Platform's [curated databases](../data/curated-databases) in a standardised way.
 
-### [GitHub](https://github.com/)
+## Integrated Development Environment (IDE)
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/tools/index.md:32]:
```suggestion
## Integrated Development Environments (IDE)
```

File: source/documentation/tools/index.md [hunk 10]
```diff
@@ -73,7 +78,7 @@ Queries MoJAP athena databases with features such as temp table creation.
 ### [splink](https://github.com/moj-analytical-services/splink)
 Provides the ability to link datasets at scale. Splink is the matching engine behind the linked data on the Analytical Platform. This package is maintained by the Internal Data Linking team, support is offered via the **#ask-data-linking** Slack channel.
 
-## R packages
+## R Packages
 
 The following R packages are maintained the AP community
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/tools/index.md:83]:
```suggestion
The following R packages are maintained by the AP community
```

File: source/documentation/data-docs/data-faqs/index.md [hunk 11]
```diff
@@ -32,18 +44,22 @@ Databases on the AP use Amazon Athena which allow you to query data using SQL. Y
...
 
-## I am running into memory issues, what should I do?
+## I am running into memory issues using Python/R, what should I do?
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/data-docs/data-faqs/index.md:55]:
```suggestion
## What should I use to process my data?

Please refer to [tools for processing my data](#link)

## I am running into memory issues using Python/R, what should I do?
```

File: source/documentation/tools/tools-for-data-processing.md [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+Another option is to use the [Athena Console](https://user-guidance.analytical-platform.service.justice.gov.uk/data/curated-databases/amazon-athena/) directly on AWS through the Analytical Platform Control Panel. The Athena Console allows you to write and run SQL code directly. This is very handy when exploring data that is already in an Athena database and table.
+
+> ✅ Pros
+> * Quick to get started as you don’t need to install anything
+> * Creates modular SQL script thinking needed for [Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) (see below)
+
+> ❌ Cons
+> * SQL only - limiting how complex your processing can be
+> * No proper version control, makes it hard for others to replicate
+> * Data has to be in an Athena database already
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/tools/tools-for-data-processing.md:45]: Not necessarily

File: source/documentation/tools/tools-for-data-processing.md [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+> ✅ Pros
+> * Quick to get started as you don’t need to install anything
+> * Creates modular SQL script thinking needed for [Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) (see below)
+
+> ❌ Cons
+> * SQL only - limiting how complex your processing can be
+> * No proper version control, makes it hard for others to replicate
+> * Data has to be in an Athena database already 
+
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/tools/tools-for-data-processing.md:46]: How about adding a section on create-a-derived-table? You don't have to use for orchestration, it also makes it easier to run one-off queries:

```suggestion
 ### Create-a-derived-table
Create-a-derived-table is an in-house developed service that brings  [dbt](https://docs.getdbt.com/docs/introduction), Git, and data access controls together to allow you to deploy derived and processed tables from data available on the Analytical Platform. This extends SQL with features commonly associated with programming languages, enabling more flexible, scalable, and maintainable code. dbt can also assure the quality of your transformations through [data tests](https://docs.getdbt.com/docs/build/data-tests), for example to check whether columns contain null values.
```

File: source/documentation/tools/tools-for-data-processing.md [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+> ❌ Cons
+> * Have to set up Dockerfile and DAG
+> * Have to describe task order in DAG manually
+> * Difficult to trigger run only when underlying data has been updated - usually have to hard-code a start time
+> * Debugging and logging aren’t great
+> * Setup even for simple tasks can be complex
+
+### Create a Derived Table
+[Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) is an in-house developed service that brings dbt, Git, and data access controls together to allow you to deploy derived and processed tables from data available on the Analytical Platform; straight to Athena, in a reproducible way, and with scheduled table updates.
```
Comment (SoumayaMauthoorMOJ) [inline source/documentation/tools/tools-for-data-processing.md:66]:
```suggestion
[Create a Derived Table](https://user-guidance.analytical-platform.service.justice.gov.uk/tools/create-a-derived-table/) also allows you to run scheduled table updates.
```

**Review by SoumayaMauthoorMOJ — COMMENTED** [review]
Made a few more comments :-) Also I was wondering if perhaps you should move "Tools for Data Processing" to the data faq section instead? It makes the structure more tidy I think:
image
But happy either way

File: source/documentation/data-docs/data-faqs/index.md (hunk 7 above)
Comment (mshodge) [reply source/documentation/data-docs/data-faqs/index.md:53]: That text comes from the existing text on the guidance: https://user-guidance.analytical-platform.service.justice.gov.uk/data/data-faqs/#how-do-i-query-a-database-on-the-platform

I'll update to remove reference to 'old' and 'new'

File: source/documentation/tools/tools-for-data-processing.md (hunk 12 above)
Comment (mshodge) [reply source/documentation/tools/tools-for-data-processing.md:45]: Which bit? All of it have caveats do they?

**Comment by mshodge** [comment]
> Made a few more comments :-) Also I was wondering if perhaps you should move "Tools for Data Processing" to the data faq section instead? It makes the structure more tidy I think: image But happy either way

Yeah I'm not 100% sure either way @SoumayaMauthoorMOJ - depends where people would naturally look for it; which may be under tools more than data. If we don't want it to be at the folder level of tools though how about we just add it to a new subfolder?

File: source/documentation/data-docs/data-faqs/index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mshodge) [reply source/documentation/data-docs/data-faqs/index.md]: @SoumayaMauthoorMOJ can we resolve this one now?

File: source/documentation/tools/tools-for-data-processing.md (hunk 12 above)
Comment (SoumayaMauthoorMOJ) [reply source/documentation/tools/tools-for-data-processing.md:45]: You can create databases and tables using Athena e.g. [create table](https://docs.aws.amazon.com/athena/latest/ug/create-table.html)

**Comment by SoumayaMauthoorMOJ** [comment]
> Yeah I'm not 100% sure either way @SoumayaMauthoorMOJ - depends where people would naturally look for it; which may be under tools more than data. If we don't want it to be at the folder level of tools though how about we just add it to a new subfolder?

Sure to be fair the whole of AP user guidance structure needs a rethink anyway so not going to solve it now :-)

**Comment by SoumayaMauthoorMOJ** [comment]
I think the PR is still trying to make changes to the create-a-derived-table docs?

image

File: source/documentation/tools/tools-for-data-processing.md (hunk 12 above)
Comment (mshodge) [reply source/documentation/tools/tools-for-data-processing.md]: Resolved by removing

File: source/tools/data-processing/tools-for-data-processing.html.md.erb [hunk 15]
```diff
@@ -0,0 +1,11 @@
+---
+title: Tools for Data Processing
+weight: 50
```
Comment (SoumayaMauthoorMOJ) [inline source/tools/data-processing/tools-for-data-processing.html.md.erb:3]:
```suggestion
weight: 5
```

**Review by SoumayaMauthoorMOJ — COMMENTED** [review]
One minor change to make sure this page is ordered  below tools and before create-a-derived-table

**Pull request merged**